src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureCosmosDBProvisioner.cs

```diff
@@ -94,6 +94,17 @@ public override bool ConfigureResource(IConfiguration configuration, AzureCosmos
         var connectionStrings = context.UserSecrets.Prop("ConnectionStrings");
         connectionStrings[resource.Name] = resource.ConnectionString;
 
+        var existingDatabases = cosmosResource.GetCosmosDBSqlDatabases().GetAllAsync(cancellationToken);
+
+        await foreach (var existingDatabase in existingDatabases)
+        {
+            if (!resource.Databases.Any(d => d.Name == existingDatabase.Data.Name))
```

Comment:
You could do a single pass over the set of databases but I'm not concerned about performance 😄